api/tests/test_routes/test_auth.py

```diff
@@ -0,0 +1,85 @@
+import json
+from utils.constants import USER_TEST_DATA, AUTH_ERRORS
+from utils.requests import user_login
+
+
+def test_auth(client):
+    response = client.post("/users/", json.dumps(USER_TEST_DATA))
+    response_json = response.json()
+    assert USER_TEST_DATA["email"] == response_json["email"]
+    assert USER_TEST_DATA["firstname"] == response_json["firstname"]
+    assert USER_TEST_DATA["lastname"] == response_json["lastname"]
+
+    response = user_login(client)
+    assert 200 == response.status_code
+    response_data = response.json()
+    assert "access_token" in response_data
+    assert "refresh_token" in response_data
+
+
+def test_auth_error_password_not_mutch_email(client):
+    response = client.post("/users/", json.dumps(USER_TEST_DATA))
+    response_json = response.json()
+    assert USER_TEST_DATA["email"] == response_json["email"]
+    assert USER_TEST_DATA["firstname"] == response_json["firstname"]
+    assert USER_TEST_DATA["lastname"] == response_json["lastname"]
+
+    response = user_login(client, password="wrong_password")
+    assert response.status_code == 401
+    assert AUTH_ERRORS["password_not_mutch_email"] == response.json()["detail"]
+
+
+def test_auth_error_email_not_exist(client):
+    response = client.post("/users/", json.dumps(USER_TEST_DATA))
```

Comment:
In general you should use fixtures to fill the db or mock crud functions, because if your create user endpoint will be broken one day, all other test fail.

Reply:
Ok, I got it I will use it in the feature